optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/solution/descriptor/SolutionDescriptor.java

```diff
@@ -617,33 +639,38 @@ public int countReinitializableVariables(ScoreDirector scoreDirector, Solution s
         List<Iterator<Object>> iteratorList = new ArrayList<Iterator<Object>>(
                 entityPropertyAccessorMap.size() + entityCollectionPropertyAccessorMap.size());
         for (MemberAccessor entityMemberAccessor : entityPropertyAccessorMap.values()) {
-            Object entity = extractEntity(entityMemberAccessor, solution);
+            Object entity = extract(entityMemberAccessor, solution);
             if (entity != null) {
                 iteratorList.add(Collections.singletonList(entity).iterator());
             }
         }
         for (MemberAccessor entityCollectionMemberAccessor : entityCollectionPropertyAccessorMap.values()) {
-            Collection<Object> entityCollection = extractEntityCollection(entityCollectionMemberAccessor, solution);
+            Collection<Object> entityCollection = extractCollection(entityCollectionMemberAccessor, solution);
             iteratorList.add(entityCollection.iterator());
         }
         return Iterators.concat(iteratorList.iterator());
     }
 
-    private Object extractEntity(MemberAccessor entityMemberAccessor, Solution solution) {
-        return entityMemberAccessor.executeGetter(solution);
+    private Object extract(MemberAccessor memberAccessor, Solution solution) {
```

Comment:
This method should probably move to ```MemberAccessor``` itself. It has no business being on ```SolutionDescriptor``` and may be made static.